distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1083,6 +1089,158 @@ type Table
+           type that is incompatible with the type coming from the first table,
+           a `Column_Type_Mismatch` is reported. The problematic column will be
+           dropped from the resulting table. With type widening disabled, the
+           subsequent tables must have the same types as the first one, unless
+           the type of the first one was `Mixed` which will accept any other
+           type.
+         - If a common type coercion for a set of matched columns from
+           concatenated tables cannot be found, a `No_Common_Type` is reported.
+           In warning or ignore mode, the problematic column will be dropped
+           from the resulting table.
+
+       ? Unifying Column Types
+
+         If `allow_type_widening` is set to `True`, then the following rules are
+         used to find a common type that will fit values from all merged tables.
+
+         Numeric columns are unified by finding the most general type that can
+         fit all of the columns. The biggest integer type will be chosen and if
+         integers and decimals are mixed, the decimal type will be chosen.
+         If boolean columns are mixed with numeric columns, they will be coerced
+         to the numeric type (and converted to 0 and 1).
+
+         Text types will also be coerced according to the common rules - if
+         constant-length texts of different lengths are mixed, they will be
+         coerced to a varying-length type.
+
+         If one of the matched columns has `Mixed` type, that type will be used
+         regardless of types of other columns. Mixing any other types will
+         result in a `No_Common_Type` problem. If columns of incompatible types
+         are meant to be mixed, at least one of them should be explicitly
+         retyped to the `Mixed` type to indicate that intention.
+    union : (Table | Vector Table) -> Match_Columns -> Boolean | Report_Unmatched -> Boolean -> Problem_Behavior -> Table
+    union self tables match_columns=Match_Columns.By_Name keep_unmatched_columns=Report_Unmatched allow_type_widening=True on_problems=Report_Warning =
+        tables_vector = case tables of
+            v : Vector -> v
+            single_table -> [single_table]
+        is_everything_ok = tables_vector.all table->
+            if Table_Helpers.is_table table . not then Error.throw (Type_Error.Error Table table "tables") else
+                same_backend = case table of
+                    _ : Table -> True
+                    _ -> False
+                if same_backend . not then Error.throw (Illegal_Argument.Error "Currently cross-backend union is not supported. Materialize the table using `.read` before merging it with an in-memory Table.") else
+                    True
+        ## `is_everything_ok` should actually never be False, it will either be
+           True or will contain a dataflow error which will propagate through
+           the if statement.
+        if  is_everything_ok then
```

Comment:
```suggestion

        ## `is_everything_ok` should actually never be False; it will either be
           True or will contain a dataflow error propagating through the if.
        if is_everything_ok then
```

Reply:
A little extra white space to make it clearer to read.